src/bam/mod.rs

```diff
@@ -438,6 +452,12 @@ quick_error! {
         NoMoreRecord {
             description("no more record")
         }
+        FetchError {
+            description("bam fetch failed")
+        }
+        SeekError {
+            description("bam seek failed")
+        }
```

Comment:
not sure what the general principle is, but they are separate kinds of failures.  the fetch can fail if there's a problem with the index, or you pass an unrecongized contig.  the seek can fail if you pass an invalid voffset. Not sure if it's worth having separate errors or not.